postgres/datadog_checks/postgres/statements.py

```diff
@@ -164,35 +164,58 @@ def collect_per_statement_metrics(self, db, db_version, tags):
             self._log.exception('Unable to collect statement metrics due to an error')
             return []
 
-    def _load_pg_stat_statements(self, db):
-        available_columns = set(self._get_pg_stat_statements_columns(db))
-        missing_columns = PG_STAT_STATEMENTS_REQUIRED_COLUMNS - available_columns
-        if len(missing_columns) > 0:
-            self._log.warning(
-                'Unable to collect statement metrics because required fields are unavailable: %s',
-                ', '.join(list(missing_columns)),
+    def _load_pg_stat_statements(self, db, tags):
+        try:
+            available_columns = set(self._get_pg_stat_statements_columns(db))
+            missing_columns = PG_STAT_STATEMENTS_REQUIRED_COLUMNS - available_columns
+            if len(missing_columns) > 0:
+                self._log.warning(
+                    'Unable to collect statement metrics because required fields are unavailable: %s',
+                    ', '.join(list(missing_columns)),
+                )
+                return []
+
+            query_columns = sorted(list(available_columns & PG_STAT_ALL_DESIRED_COLUMNS))
+            params = ()
+            filters = ""
+            if self._config.dbstrict:
+                filters = "AND pg_database.datname = %s"
+                params = (self._config.dbname,)
+            return self._execute_query(
+                db.cursor(cursor_factory=psycopg2.extras.DictCursor),
+                STATEMENTS_QUERY.format(
+                    cols=', '.join(query_columns),
+                    pg_stat_statements_view=self._config.pg_stat_statements_view,
+                    filters=filters,
+                    limit=DEFAULT_STATEMENTS_LIMIT,
+                ),
+                params=params,
             )
+        except psycopg2.errors.ObjectNotInPrerequisiteState as e:
+            if 'pg_stat_statements must be loaded' in str(e):
+                self._check.count(
+                    "dd.postgres.statement_metrics.error", 1, tags=tags + ["error:pg_stat_statements_not_enabled"]
```

Comment:
It may be more consistent to have this error tag be something like `error:database-ObjectNotInPrerequisiteState(pg_stat_statements_not_loaded)` so they all follow that same pattern `error:database-{exception_type}.*`

Reply:
That's a good suggestion. Done ✅ .